cpp/src/io/parquet/writer_impl.cu

```diff
@@ -521,62 +464,80 @@ struct leaf_schema_fn {
   template <typename T>
   std::enable_if_t<std::is_same_v<T, cudf::duration_s>, void> operator()()
   {
-    col_schema.type           = Type::INT32;
-    col_schema.converted_type = ConvertedType::TIME_MILLIS;
-    col_schema.stats_dtype    = statistics_dtype::dtype_int32;
-    col_schema.ts_scale       = 1000;
-    col_schema.logical_type   = LogicalType{TimeType{timestamp_is_utc, TimeUnit::MILLIS}};
+    col_schema.type = (write_arrow_schema) ? Type::INT64 : Type::INT32;
+    col_schema.stats_dtype =
+      (write_arrow_schema) ? statistics_dtype::dtype_int64 : statistics_dtype::dtype_int32;
+    // only write as time32 logical type if not writing arrow schema
+    if (not write_arrow_schema) {
+      col_schema.converted_type = ConvertedType::TIME_MILLIS;
+      col_schema.logical_type   = LogicalType{TimeType{timestamp_is_utc, TimeUnit::MILLIS}};
+      col_schema.ts_scale       = 1000;
+    }
```

Comment:
```suggestion
    // if writing arrow schema, no logical type nor converted type is necessary
    if (write_arrow_schema) {
      col_schema.type        = Type::INT64;
      col_schema.stats_dtype = statistics_dtype::dtype_int64;
    } else {
      // parquet TIME_MILLIS annotates INT32
      col_schema.type           = Type::INT32;
      col_schema.stats_dtype    = statistics_dtype::dtype_int32;
      col_schema.converted_type = ConvertedType::TIME_MILLIS;
      col_schema.logical_type   = LogicalType{TimeType{timestamp_is_utc, TimeUnit::MILLIS}};
      col_schema.ts_scale       = 1000;
    }
```
Not sure if it's a little cleaner looking getting rid of the multiple ternary operators (same comment below for `duration_ms`).

Reply:
I don't mind either or. I guess we can ask folks for their preference here. @vuule?

Reply:
not a strong preference, but the single if is easier to follow